### PR TITLE
Settings Refactor

### PR DESCRIPTION
**Pull request by jguerinet**
* Cleaned up all code that is accessible from the Settings except for the walkthrough (settings, about, help, EULA)
* Clean up the about section by removing the roles under each person
* Removed useless FAQs 
* Switched the beta tester Url to the one from the Google Play Store
* Switched some of the layouts to ConstraintLayout

### CODE REVIEW

File: app/src/main/java/ui/settings/about/PersonAdapter.kt [hunk 1]
```diff
@@ -109,80 +75,92 @@ class PersonAdapter internal constructor() : BaseRecyclerViewAdapter(), KoinComp
...
+                    picture.visibility = View.GONE
+                }
 
-                role.setText(person.role)
-
-                linkedIn.setOnClickListener {
-                    ga.sendEvent("About", "Linkedin", context.getString(person.name))
-                    context.openUrl(context.getString(person.linkedIn))
+                // LinkedIn
+                val linkedInId = getResourceId(idPrefix + "_linkedin")
```
Comment (AllanWang) [inline app/src/main/java/ui/settings/about/PersonAdapter.kt]: For the most part, it's recommended that you don't concatenate strings. You'll probably get a warning about this in lint as well. In Kotlin, wrap variables with `${...}`. So this becomes `${idPrefix}_linkedin`. In cases where  only a variable is wrapped and the proceeding characters aren't letters, you. Ktlint should actually do that automatically when it's added

File: app/src/main/res/layout/item_person.xml [hunk 2]
```diff
@@ -1,75 +1,72 @@
...
-</LinearLayout>
\ No newline at end of file
+        android:layout_marginEnd="8dp"
+        android:layout_marginTop="8dp"
+        android:background="?android:attr/selectableItemBackground"
+        android:contentDescription="@null"
+        android:padding="@dimen/padding_micro"
+        android:tint="@color/red"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toEndOf="@+id/linkedIn"
```
Comment (AllanWang) [inline app/src/main/res/layout/item_person.xml]: Lowercase `i` to remain consistent?

File: app/src/main/java/ui/settings/about/PersonAdapter.kt (hunk 1 above)
Comment (jguerinet) [reply app/src/main/java/ui/settings/about/PersonAdapter.kt]: Weird, I normally don't do this. I must be tired...

File: app/src/main/res/layout/item_person.xml (hunk 2 above)
Comment (jguerinet) [reply app/src/main/res/layout/item_person.xml]: It should actually all be `linkedIn` or `linked_in` if uppercases aren't available. So the String Id should change here, not this.

**Comment by jguerinet** [comment]
With regards to your note: 
* I haven't set up ProGuard yet, it's a bridge to cross in the near future. 
* It was originally using enums with fully resolved Ids, but given that it's so repetitive and follows the same exact format, I switched to this in this PR. When ProGuard becomes a thing, if  this is problematic I'll revert back to using enums.

**Comment by AllanWang** [comment]
enum would be something like this:

```
enum class Person(val aboutRes: Int, val iconRes: Int, ...) {
    JULIEN(R.string.about_julien, R.drawable.about_julien, R.string.about_julien_role,
                R.string.about_julien_email, R.string.about_julien_linkedin),
    XYZ(...)
}
```

and you would add items through `Person.values().map {...}`.

If you have subcategories you can add extra values to denote the categories. It isn't too verbose in the sense that every value you write has meaning, and it's safer than getting resources by name.

**Comment by jguerinet** [comment]
All right all right, I'll add the enums back.

File: app/src/main/java/ui/settings/about/PersonAdapter.kt [hunk 3]
```diff
@@ -109,80 +75,137 @@ class PersonAdapter internal constructor() : BaseRecyclerViewAdapter(), KoinComp
...
     /**
      * Header in the list
      */
     internal inner class HeaderHolder(private val view: TextView) :
-            BaseRecyclerViewAdapter.BaseHolder(view) {
+        BaseRecyclerViewAdapter.BaseHolder(view) {
 
         override fun bind(position: Int) {
-            view.setText(items[position] as Int)
+            val title = items[position] as? Int ?: return
```
Comment (AllanWang) [inline app/src/main/java/ui/settings/about/PersonAdapter.kt]: You may consider this a safer binding, but having the item not be an int is unexpected behaviour

File: app/src/main/java/ui/settings/about/PersonAdapter.kt [hunk 4]
```diff
@@ -109,80 +75,137 @@ class PersonAdapter internal constructor() : BaseRecyclerViewAdapter(), KoinComp
...
-
-                role.setText(person.role)
+                    .load(person.pictureRes)
+                    .into(picture)
 
+                // LinkedIn
                 linkedIn.setOnClickListener {
-                    ga.sendEvent("About", "Linkedin", context.getString(person.name))
-                    context.openUrl(context.getString(person.linkedIn))
+                    ga.sendEvent("About", "Linkedin", person.name)
```
Comment (AllanWang) [inline app/src/main/java/ui/settings/about/PersonAdapter.kt]: Capital i in `Linkedin`? (The hardcoded string)

File: app/src/main/java/ui/settings/about/PersonAdapter.kt (hunk 3 above)
Comment (jguerinet) [reply app/src/main/java/ui/settings/about/PersonAdapter.kt]: I'm not sure what you're getting at here. Would you rather error out if this is the case, i.e. 
```suggestion
            val title = items[position] as? Int ?: error("Not an Int")
```

Comment (AllanWang) [reply app/src/main/java/ui/settings/about/PersonAdapter.kt]: Given previous checks, you should never have a viewholder with an incorrectly associated data type. This way of handling prevents a crash, but doesn't actually give a desired solution when the cast fails, as it'll mess up the UI.

You can leave this for now, but in the future we can look into creating a different type of adapter where the layoutRes, viewholder, and data are all held in a single model. That way you can deal with whatever data you wish without needing to cast. An example of such an adapter is [FastAdapter](https://github.com/mikepenz/FastAdapter), but you can write your own if you don't want to import another library.

Comment (jguerinet) [reply app/src/main/java/ui/settings/about/PersonAdapter.kt]: Fair enough. I'll rework the adapters I have in Suitcase hold those three pieces of data together.

**Pull request merged**